nemo/collections/nlp/data/language_modeling/megatron/gpt_dataset.py

```diff
@@ -155,6 +164,10 @@ def get_indexed_dataset_(data_prefix, data_impl, skip_warmup):
 
 class GPTDataset(MegatronDataset):
     def __init__(self, cfg, trainer, name, data_prefix, documents, indexed_dataset, num_samples, seq_length, seed):
+        if not HAVE_APEX:
+            logging.warning(
+                "Apex was not found. Please see the NeMo README for installation instructions: https://github.com/NVIDIA/NeMo#megatron-gpt."
+            )
```

Comment:
Not a warning, raise an import error. Code below this won't work without apex anyway no ?

Reply:
I don't understand this, python will you when there's an import error.

Reply:
Python will give you a generic error without information of how to install apex. And the stack traces are large, so your comment about installing apex as a warning will not be seen unless someone actively looks for it. It is better to raise an import error explicitly with the exact message of how to install it. But it's upto you

Reply:
Done.